### PR TITLE
Damage card basic

### PR DESCRIPTION
**Pull request by gatesvp**
Very first cut at #277.

The changeset does three basic things:

1. Add an `applyDamage` method to Actors
2. Add hook that triggers whenever a user right-clicks a menu, that hook has logic to verify which options are available and to execute `applyDamage` with the appropriate parameters
3. Carries forward additional roll information for our "attack cards", enough for introspection on the rolls we actually made

None of this is really "automation", but I think number 3 will help someone else execute automation.

Video of the feature in action. All of the related Quench tests are still passing.
https://youtu.be/kmyGdTtIEkQ

### CODE REVIEW

File: system/src/chat/hooks.mjs [hunk 1]
```diff
@@ -132,3 +132,146 @@ export default function onRenderChatMessage(app, html, data) {
...
+	}));
+}
+
+
+/**
+ * Identifies basic ChatMessage rolls like `/r d6`
+ * @param {ChatMessage} message
+ * @returns
+ */
+function _chatMessageIsBasicRoll (message) {
```
Comment (bakbakbakbakbak) [inline system/src/chat/hooks.mjs:253]: Is your linting on and properly using the config? This is throwing a warning for me if I look at it in VSCodium at least.

File: system/src/documents/ActorSD.mjs [hunk 2]
```diff
@@ -36,6 +36,30 @@ export default class ActorSD extends Actor {
 		}
 	}
 
+	/**
+	 * Applies the given number to the Actor or Token's HP value.
+	 * The multiplier is a convenience feature to apply healing
+	 *  or true multiples of a damage value.
+	 *  * 1 => damage as rolled
+	 *  * .05 => half damage (resistance)
```
Comment (bakbakbakbakbak) [inline system/src/documents/ActorSD.mjs:44]: This should be `0.5` right?

File: system/src/documents/ActorSD.mjs [hunk 3]
```diff
@@ -36,6 +36,30 @@ export default class ActorSD extends Actor {
...
+	 * @param {number} damageAmount
+	 * @param {number} multiplier
+	 */
+	async applyDamage(damageAmount, multiplier) {
+		const maxHpValue = this.system.attributes.hp.max;
+		const currentHpValue = this.system.attributes.hp.value;
+		const amountToApply = Math.floor(parseInt(damageAmount) * multiplier);
+
+		// Ensures that we don't go above Max or below Zero
+		const newHpValue = Math.clamped(currentHpValue - amountToApply, 0, maxHpValue)
```
Comment (bakbakbakbakbak) [inline system/src/documents/ActorSD.mjs:56]: See comment above on linting!

**Review by bakbakbakbakbak — CHANGES_REQUESTED** [review]
I've reviewed the changes, and there are some things to look into / comment about.

**Review by bakbakbakbakbak — APPROVED** [review]
Looks good!

**Comment by bakbakbakbakbak** [comment]
@gatesvp : As with the other PR, can you add a commit with updates to the `RELEASE_NOTES.md` and I'll merge this.

**Comment by gatesvp** [comment]
> @gatesvp : As with the other PR, can you add a commit with updates to the `RELEASE_NOTES.md` and I'll merge this.

This piece is done.

It looks like there's a conflict to resolve in the main `shadowdark.mjs` file. I think you know the other changes, will leave that to you to merge?

**Pull request merged**